packages/protocol/src/protocol-thp/crypto/crypto.ts

```diff
@@ -0,0 +1,219 @@
+    aesGcm.auth(Buffer.alloc(0));
+    const trezorState = aesGcm.decrypt(payload.subarray(0, 1), payload.subarray(1, 17));
+
+    return trezorState.readUint8();
+};
+
+type ThpCryptoOptions = {
+    handshakeHash: Buffer;
+    trezorEphemeralPubkey: Buffer;
+    trezorEncryptedStaticPubkey: Buffer;
+    tag: Buffer;
+    sendNonce: number;
+    recvNonce: number;
+    // knownCredentials: any[];
+};
+
+export const handleHandshakeInitResponse = ({
+    handshakeHash,
+    trezorEphemeralPubkey,
+    trezorEncryptedStaticPubkey,
+    tag,
+    // knownCredentials,
+    sendNonce,
+    recvNonce,
+}: ThpCryptoOptions) => {
+    const iv0 = getIvFromNonce(sendNonce); // should be 0
+    const iv1 = getIvFromNonce(recvNonce); // should be 1
+
+    const hostStaticKeys = getCurve25519KeyPair(Buffer.alloc(32).fill(9));
+    const hostEphemeralKeys = getCurve25519KeyPair(Buffer.alloc(32).fill(1));
+
+    let h: Buffer, point: Buffer, aesGcm: AESGCM;
+
+    // 1. Set h = SHA-256(protocol_name || device_properties).
+    // h = hash_of_two(PROTOCOL_NAME, deviceProperties); // moved to handleCreateChannelResponse
+    h = handshakeHash;
+    // 2. Set h = SHA-256(h || host_ephemeral_pubkey).
+    h = hashOfTwo(h, hostEphemeralKeys.publicKey);
+    // 3. Set h = SHA-256(h || trezor_ephemeral_pubkey).
+    h = hashOfTwo(h, trezorEphemeralPubkey);
+    // 4. Set ck, k = HKDF(protocol_name, X25519(host_ephemeral_privkey, trezor_ephemeral_pubkey)).
+    point = curve25519(hostEphemeralKeys.privateKey, trezorEphemeralPubkey);
+    let [ck, k] = hkdf(PROTOCOL_NAME, point);
+
+    // 5. Set trezor_masked_static_pubkey, success = AES-GCM-DECRYPT(key=k, IV=0^96 (bits, 12 bytes), ad=h, plaintext=encrypted_trezor_static_pubkey). Assert that success is True.
+    aesGcm = new AESGCM(k, iv0);
+    aesGcm.auth(h);
+    const trezorStaticPubkey = trezorEncryptedStaticPubkey.subarray(0, 32);
+    const trezorStaticPubkeyTag = trezorEncryptedStaticPubkey.subarray(32, 32 + 16);
+    const trezorMaskedStaticPubkey = aesGcm.decrypt(trezorStaticPubkey, trezorStaticPubkeyTag);
+    // 6. Set h = SHA-256(h || encrypted_trezor_static_pubkey)
+    h = hashOfTwo(h, trezorEncryptedStaticPubkey);
+    // 7. Set ck, k = HKDF(ck, X25519(host_ephemeral_privkey, trezor_masked_static_pubkey))
+    point = curve25519(hostEphemeralKeys.privateKey, trezorMaskedStaticPubkey);
+    [ck, k] = hkdf(ck, point);
+
+    // 8. Set tag_of_empty_string, success = AES-GCM-DECRYPT(key=k, IV=0^96 (bits, 12 bytes), ad=h, plaintext=empty_string). Assert that success is True.
+    aesGcm = new AESGCM(k, iv0);
+    aesGcm.auth(h);
+    aesGcm.decrypt(Buffer.alloc(0), tag);
+    // 9. Set h = SHA-256(h || tag)
+    h = hashOfTwo(h, tag);
+
+    // const credentials = knownCredentials.find(c => {
+    //     const isPaired = findKnownPairingCredentials(knownCredentials[0], trezorEphemeralPubkey);
+    // })
+    const hostTempKeys = hostStaticKeys || getCurve25519KeyPair(Buffer.alloc(32).fill(9));
```

Comment:
The bytes inside `getcurve25519KeyPair` should be `0900000000000000000000000000000000000000000000000000000000000000` instead of `0909090909090909090909090909090909090909090909090909090909090909`.